### PR TITLE
add: note to direct users to modifiyng frontmatter with remark

### LINKED ISSUE #3930: Not clear how to modify frontmatter programmatically
### 📚 Subject area/topic

Markdown

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/guides/markdown-content/#modifying-frontmatter-programmatically

### 📋 General description or bullet points (if proposing new content)

I was trying to add a “modified time” to the bottom of the MDX articles on my website. I found the [modifying frontmatter programmatically section](https://docs.astro.build/en/guides/markdown-content/#modifying-frontmatter-programmatically) in the Astro docs.

So I created a simple remark plugin `modifiedTime.mjs` that gets a file's modified time from git:
```js
import { execSync } from "child_process";

export function modifiedTime() {
  return function (tree, file) {
    const filepath = file.history[0]; // Not sure this is the best place to get the filepath...
    const result = execSync(`git log -1 --pretty="format:%cI" ${filepath}`);
    file.data.astro.frontmatter.lastModified = result.toString();
  };
}
```

I then modified my Astro config, as per the documentation:
```js
import { defineConfig } from "astro/config";

import mdx from "@astrojs/mdx";
import { modifiedTime } from "./src/plugins/modifiedTime.mjs";

// https://astro.build/config
export default defineConfig({
  integrations: [mdx()],
  markdown: {
    remarkPlugins: [modifiedTime],
  },
});
```

And finally, in my page that renders the markdown (simplified):
```astro
---
import { getCollection } from "astro:content";

import Layout from "../../layouts/WikiArticle.astro";
import type { CollectionEntry } from "astro:content";
import dayjs from "../../services/time";

export const getStaticPaths = async () => {
  const articles = await getCollection("wiki");
  return articles.map((article) => {
    return { params: { slug: article.slug }, props: { article } };
  });
};

const { article } = Astro.props;
const { Content, headings } = await article.render();
const slug = article.data.title.toLowerCase().replaceAll(" ", "-");

export interface Props {
  article: CollectionEntry<"wiki">;
}

const modifiedTime = article.data.lastModified ?? "???";
---

<Layout article={article} headings={headings}>
  <Content  />
  <div class="modified-timestamp">
    <p>Last Modified: {modifiedTime}</p>
  </div>
</Layout>
```
Note that in this page, I am using `getStaticPaths()`, which I learned from the [content-collections page](https://docs.astro.build/en/guides/content-collections/#building-for-static-output-default). This is where I got lost. Because my content is in MDX files, I was reading the [MDX solution](https://docs.astro.build/en/guides/markdown-content/#modifying-frontmatter-programmatically) - but after hours of not being able to figure out why it wasn't working, I discovered that I actually need to follow [modifying frontmatter with remark](https://docs.astro.build/en/guides/content-collections/#modifying-frontmatter-with-remark).

Using the below now works:
```diff
---
import { getCollection } from "astro:content";

import Layout from "../../layouts/WikiArticle.astro";
import type { CollectionEntry } from "astro:content";
import dayjs from "../../services/time";

export const getStaticPaths = async () => {
  const articles = await getCollection("wiki");
  return articles.map((article) => {
    return { params: { slug: article.slug }, props: { article } };
  });
};

const { article } = Astro.props;
- const { Content, headings } = await article.render();
+ const { Content, headings, remarkPluginFrontmatter } = await article.render();
const slug = article.data.title.toLowerCase().replaceAll(" ", "-");

export interface Props {
  article: CollectionEntry<"wiki">;
}

- const modifiedTime = article.data.lastModified ?? "???";
+ const modifiedTime = remarkPluginFrontmatter.lastModified ?? "???"
---

<Layout article={article} headings={headings}>
  <Content  />
  <div class="modified-timestamp">
    <p>Last Modified: {modifiedTime}</p>
  </div>
</Layout>
```

I spent hours trying to get this to work because, oddly enough, sometimes it **DID** work. When saving the `astro.config.mjs` file, vite would restart and the current page I was on would actually **correctly** display the modified time at the bottom of the page. However, killing the dev server with `CTRL + C` and running `npm run dev` again (or refreshing the page) does **not** work, with the modified time being ??? (`undefined`)

### 🖥️ Reproduction in StackBlitz (if reporting incorrect content or code samples)

It isn't incorrect, just not clear. There should be some sort of warning in the https://docs.astro.build/en/guides/markdown-content/#modifying-frontmatter-programmatically section perhaps that directs users to https://docs.astro.build/en/guides/content-collections/#modifying-frontmatter-with-remark

### PR DESCRIPTION
**Pull request by carsakiller**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)
- New or updated content

#### Description

- Closes #3930
- What does this PR change? Give us a brief description.
  - It adds a note `aside` to the `markdown-content` page that warns users that if they are using content collections, they should instead be reading the `content-collections` page if they want to modify their frontmatter programmatically.

### CODE REVIEW

**Comment by carsakiller** [comment]
[Direct link](https://deploy-preview-3988--astro-docs-2.netlify.app/en/guides/markdown-content/#modifying-frontmatter-programmatically) to the note I have added

File: src/content/docs/en/guides/markdown-content.mdx [hunk 1]
```diff
@@ -515,6 +515,10 @@ export default {
 
 ### Modifying frontmatter programmatically
 
+:::note
+If you are using [content collections](/en/guides/content-collections/), you will instead want to read ["Modifying Frontmatter with Remark"](/en/guides/content-collections/#modifying-frontmatter-with-remark)
```
Comment (sarah11918) [inline src/content/docs/en/guides/markdown-content.mdx:519]:
```suggestion
If you are using [content collections](/en/guides/content-collections/), please see ["Modifying Frontmatter with Remark"](/en/guides/content-collections/#modifying-frontmatter-with-remark).
```
A little more direct!

**Review by sarah11918 — APPROVED** [review]
Thank you so much for first filing a really helpful, detailed issue and then jumping on the fix, @carsakiller! :rocket: 

I kind of had to laugh when I *followed* your link, and got a huge DO NOT DO THIS warning, but... I mean, this *is* the proper place to direct people, so... :joy:

I'm just going to commit the tiny tweak to the language directly here, and then we'll officially make you a member of Team Docs!  :partying_face:

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks for this lovely note @carsakiller 💜

**Pull request merged**